service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java

```diff
@@ -475,6 +475,19 @@ public CourierDto deactivateCourier(Long id) {
         return modelMapper.map(courier, CourierDto.class);
     }
 
+    @Override
+    @Transactional
+    public CourierDto activateCourier(Long id) {
+        Courier courier = courierRepository.findById(id).orElseThrow(
+            () -> new NotFoundException(ErrorMessage.COURIER_IS_NOT_FOUND_BY_ID + id));
+        if (CourierStatus.ACTIVE == courier.getCourierStatus()) {
+            throw new BadRequestException(ErrorMessage.CANNOT_ACTIVATE_COURIER + courier.getId());
+        }
+        deactivateTariffsForChosenParamRepository.deactivateTariffsByCourier(id);
```

Comment:
why should we deactivate all tariffs before activating the courier?

Reply:
fixed